File: app/src/main/java/eu/szwiec/replayview/ApplicationModule.java
package eu.szwiec.replayview;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import eu.szwiec.replayview.replay.FilesProviderImpl;

@Module
public class ApplicationModule {

    private Application application;

    public ApplicationModule(Application application) {
        this.application = application;
    }

    @Provides
    @Singleton
    public Context provideContext() {
        return application;
    }

    @Provides
    @Singleton
    public FilesProvider provideFilesProvider(Context context) {
        return new FilesProviderImpl(context);
    }

}

File: app/src/main/java/eu/szwiec/replayview/ApplicationModule.kt
package eu.szwiec.replayview

import android.app.Application
import android.content.Context

import javax.inject.Singleton

import dagger.Module
import dagger.Provides
import eu.szwiec.replayview.replay.ReplayFilesProvider
import eu.szwiec.replayview.replay.ReplayFilesProviderImpl
import eu.szwiec.replayview.replay.ReplayViewModelFactory

@Module
class ApplicationModule(private val application: Application) {

    @Provides
    @Singleton
    fun provideContext(): Context {
        return application
    }

    @Provides
    @Singleton
    fun provideFilesProvider(context: Context): ReplayFilesProvider {
        return ReplayFilesProviderImpl(context)
    }

    @Provides
    @Singleton
    fun provideReplayViewModelFactory(filesProvider: ReplayFilesProvider): ReplayViewModelFactory {
        return ReplayViewModelFactory(filesProvider)
    }

}
